pulp_container/app/models.py

```diff
@@ -154,39 +159,85 @@ def init_image_nature(self):
             if manifest.is_bootable:
+                self.nature = MANIFEST_TYPE.BOOTABLE
                 self.is_bootable = True
                 return True
             elif manifest.is_flatpak:
+                self.nature = MANIFEST_TYPE.FLATPAK
                 self.is_flatpak = True
                 return True
 
-        return False
+        return updated_nature
 
     def init_manifest_nature(self):
-        if self.is_bootable_image():
-            self.is_bootable = True
-            return True
-        elif self.is_flatpak_image():
-            self.is_flatpak = True
-            return True
-        else:
-            return False
+        for manifest_type, check_type_function in self.known_types().items():
+            if check_type_function():
+                self.nature = manifest_type
+                return True
+
+        return False
+
+    def known_types(self):
+        return {
+            MANIFEST_TYPE.BOOTABLE: self.is_bootable_image,
+            MANIFEST_TYPE.FLATPAK: self.is_flatpak_image,
+            MANIFEST_TYPE.HELM: self.is_helm_image,
+            MANIFEST_TYPE.SIGNATURE: self.is_cosign,
+            MANIFEST_TYPE.IMAGE: self.is_manifest_image,
+        }
 
     def is_bootable_image(self):
         if (
             self.annotations.get("containers.bootc") == "1"
             or self.labels.get("containers.bootc") == "1"
         ):
+            # DEPRECATED: is_bootable is deprecated and will be removed in a future release.
+            self.is_bootable = True
             return True
         else:
             return False
 
     def is_flatpak_image(self):
-        return True if self.labels.get("org.flatpak.ref") else False
+        if self.labels.get("org.flatpak.ref"):
+            # DEPRECATED: is_flatpak is deprecated and will be removed in a future release.
+            self.is_flatpak = True
+            return True
+        return False
+
+    def is_manifest_image(self):
+        return self.media_type in (MEDIA_TYPE.MANIFEST_OCI, MEDIA_TYPE.MANIFEST_V2)
+
+    def validate_json_field(field):
+        def decorator(func):
+            @wraps(func)
+            def wrapper(self, *args, **kwargs):
+                json_manifest = json.loads(self.data)
+                if not json_manifest.get(field, None):
+                    return False
+                return func(self, json_manifest, *args, **kwargs)
+
+            return wrapper
+
+        return decorator
+
+    @validate_json_field("layers")  # Check for 'layers' field for cosign images
+    def is_cosign(self, json_manifest):
+        return any(
+            layer.get("mediaType", None) == MEDIA_TYPE.COSIGN for layer in json_manifest["layers"]
+        )
+
+    @validate_json_field("config")  # Check for 'config' field for helm images
+    def is_helm_image(self, json_manifest):
+        return json_manifest.get("config").get("mediaType") == MEDIA_TYPE.HELM
```

Comment:
So `manifest.data` needs to be a string field, because different formatting would bust the manifest checksums, right? But we can safely assume it contains valid json.
Let me suggest a different (maybe more pythonic) approach here:
We can add a `json_manifest` property.
```python
@property
def json_manifest(self):
    return json.loads(self.data)
```

Then we can do:
```python
def is_helm_image(self):
    # should this actually be `is_helm_chart`?
    try:
        return self.json_manifest["config"]["mediaType"] == MEDIA_TYPE.HELM
    except KeyError:
        return False
```

Reply:
Or just
```python
def is_helm_image(self):
    # should this actually be `is_helm_chart`?
    try:
        return json.loads(self.data)["config"]["mediaType"] == MEDIA_TYPE.HELM
    except KeyError:
        return False
```
in the first place.